pkg/odo/cli/telemetry/telemetry.go

```diff
@@ -0,0 +1,80 @@
+package telemetry
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/openshift/odo/pkg/odo/genericclioptions"
+
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/preference"
+	"github.com/openshift/odo/pkg/segment"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"k8s.io/klog"
+)
+
+const RecommendedCommandName = "telemetry"
+
+type TelemetryOptions struct {
+	telemetryData segment.TelemetryData
+}
+
+func NewTelemetryOptions() *TelemetryOptions {
+	return &TelemetryOptions{}
+}
+
+func (o *TelemetryOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	err = json.Unmarshal([]byte(args[0]), &o.telemetryData)
+	return err
+}
+
+func (o *TelemetryOptions) Validate() (err error) {
+	return err
+}
+
+func (o *TelemetryOptions) Run() (err error) {
+	var segmentClient *segment.Client
+	cfg, err := preference.New()
+	if err != nil {
+		return errors.Errorf("unable to update data, required preference.yaml file not found.")
+	}
+	// Initiate the segment client if ConsentTelemetry preference is set to true
+	if cfg.GetConsentTelemetry() {
+		if os.Getenv(segment.DisableTelemetryEnv) == "true" {
```

Comment:
I think we can use `IsTelemetryEnabled` here. We can move the warning message below inside `IsTelemetryEnabled` and also convert it into a `klog` message instead of a warning type output.